src/utils/network/session-core.js

```diff
@@ -534,5 +559,37 @@ export const createOutbox = async (session) => {
   const outboxContainerUrl = getContainerUrl(session, 'Outbox', INTERACTION_TYPES.SELF);
   await createContainerAt(outboxContainerUrl, { fetch: session.fetch });
 
-  await setDocAclForUser(session, outboxContainerUrl, 'create', session.info.webId);
+  try {
+    await getSolidDataset(outboxContainerUrl, { fetch: session.fetch });
+  } catch {
+    await createContainerAt(outboxContainerUrl, { fetch: session.fetch });
+
+    // Generate ACL file for container
+    await setDocAclForUser(session, outboxContainerUrl, 'create', session.info.webId);
+  }
+};
+
+/**
+ * Function that creates an inbox container in the user's Pod when logging in for
+ * the first time
+ *
+ * @memberof utils
+ * @function createInbox
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @returns {Promise} Promise - Generates an outbox for Pod upon log in if
+ * user's Pod does not have the an outbox to begin with
+ */
+
+export const createInbox = async (session) => {
+  const inboxContainerUrl = getContainerUrl(session, 'Inbox', 'self-fetch');
```

Comment:
Now that we have the signed-in user's pod url, it would be better to pass that in and generate inbox Url off of that.

Reply:
Yeah, I think we could start replacing these functionsl with podUrl